File: backend/music/src/main/java/com/springboot/music/dto/AdminOrderWithDetailsDTO.java
package com.springboot.music.dto;

import lombok.*;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AdminOrderWithDetailsDTO {

    private Integer orderId;
    private Integer userId;
    private String customerName;
    private String customerEmail;
    private Double totalAmount;
    private String paymentStatus;
    private String paymentMethod;
    private String gatewayTransactionCode;
    private LocalDateTime createdAt;
    private Integer totalItems;
    private List<AdminOrderDetailDTO> items;
}

File: frontend/ReactMusicCommerce/src/responsemodel/AdminOrderWithDetailsDTO.ts
import type { AdminOrderDetailDTO } from "../models/AdminOrderDetailDTO";

export interface AdminOrderWithDetailsDTO {
  orderId: number;
  userId: number;
  customerName: string;
  customerEmail: string;
  totalAmount: number;
  paymentStatus: string;
  paymentMethod: string;
  gatewayTransactionCode: string | null;
  createdAt: string;
  totalItems: number;
  items: AdminOrderDetailDTO[];
}
